lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -3333,6 +3337,33 @@ static void prepareHWModule(Block &block, ModuleNameManager &names) {
       names.addLegalName(op.getResult(0), regOp.name(), &op);
     else if (auto interfaceInstanceOp = dyn_cast<InterfaceInstanceOp>(op))
       names.addLegalName(op.getResult(0), interfaceInstanceOp.name(), &op);
+
+    // Force any expression used in the event control of an always process to be
+    // a trivial wire, if the corresponding option is set.
+    if (!options.allowExprInEventControl) {
+      auto enforceWire = [&](Value expr) {
+        auto definingOp = expr.getDefiningOp();
+        if (!definingOp || isa<WireOp, ReadInOutOp>(definingOp))
```

Comment:
I think you need something like this (to avoid reads from mems), also you should never have a direct use of a wire because it'd be an inout:

`if (!definingOp || !isa<ReadInOutOp>(definingOp) || !isa<WireOp>(cast<ReadInOutOp>(definingOp)->getOperand())`

Reply:
Ah yeah, good thinking. Updated this condition to only skip on the `read_inout(wire)` pattern.